src/Functions/seriesPeriodDetect.cpp

```diff
@@ -69,51 +78,53 @@ class FunctionSeriesPeriodDetect : public IFunction
     template <typename T>
     bool executeNumber(const IColumn & src_data, Float64 & period) const
     {
-        if (const ColumnVector<T> * src_data_concrete = checkAndGetColumn<ColumnVector<T>>(&src_data))
-        {
-            const PaddedPODArray<T> & src_vec = src_data_concrete->getData();
+        const ColumnVector<T> * src_data_concrete = checkAndGetColumn<ColumnVector<T>>(&src_data);
+        if (!src_data_concrete)
+            return false;
+
+        const PaddedPODArray<T> & src_vec = src_data_concrete->getData();
+
+        size_t len = src_vec.size();
+        if (len < 4)
```

Comment:
Can there be a case with too many arguments?

Reply:
It should be able to accept the maximum array that can be passed to CH functions. Overall, the method accepts only one 1 argument (specified in `getNumberOfArguments`).